pkg/loadbalancerservice/controller.go

```diff
@@ -0,0 +1,206 @@
+package loadbalancerservice
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/cloud-provider/service/helpers"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/microshift/pkg/config"
+)
+
+const defaultInformerResyncPeriod = 10 * time.Minute
+
+type LoadbalancerServiceController struct {
+	NodeName   string
+	NodeIP     string
+	KubeConfig string
+	client     *kubernetes.Clientset
+	indexer    cache.Indexer
+	queue      workqueue.RateLimitingInterface
+	informer   cache.SharedIndexInformer
+}
+
+func NewLoadbalancerServiceController(cfg *config.MicroshiftConfig) *LoadbalancerServiceController {
+	return &LoadbalancerServiceController{
+		NodeIP:     cfg.NodeIP,
```

Comment:
This is a follow up question from our meeting and relates to pluggability of CNI (I didn't had time to look at this PR yet, so take it with grain of salt):
Can we (and should we) use some K8s object to pass information from the CNI to the MicroShift?
Just like CNI is setting annotations on Pods about their iface, CIDR and stuff, we could use Node's annotation or label, or maybe some configmap if we're concerned about too open RBAC

Reply:
We already pass this down to CNI from configuration of MicroShift, so it should be equal


Also, some other components before CNI need this IP, and we pass it from MicroShift down to the components initialized before CNI... so it's MicroShift -> CNI what we have to do (as far as I understood).